drop-analytics/src/mock_impl.rs

```diff
@@ -1,37 +1,13 @@
-use crate::{FileInfo, TransferDirection, TransferInfo};
+#![allow(unused_variables)]
 
 pub struct MockImpl;
 
 impl super::Moose for MockImpl {
-    fn service_quality_initialization_init(&self, _res: Result<(), i32>) {}
-
-    fn service_quality_transfer_batch(
-        &self,
-        _transfer_id: String,
-        _info: TransferInfo,
-        _protocol_version: i32,
-    ) {
-    }
-
-    fn service_quality_transfer_file(
-        &self,
-        _res: Result<(), i32>,
-        _transfer_id: String,
-        _transfer_time: i32,
-        _direction: TransferDirection,
-        _info: Option<FileInfo>,
-    ) {
-    }
-
-    fn developer_exception(&self, _code: i32, _note: String, _message: String, _name: String) {}
-
-    fn developer_exception_with_value(
-        &self,
-        _arbitrary_value: i32,
-        _code: i32,
-        _note: String,
-        _message: String,
-        _name: String,
-    ) {
-    }
+    fn event_init(&self, data: crate::InitEventData) {}
+    fn event_transfer_intent(&self, data: crate::TransferIntentEventData) {}
+    fn event_transfer_start(&self, data: crate::TransferStartEventData) {}
+    fn event_transfer_end(&self, data: crate::TransferEndEventData) {}
+    fn event_transfer_file(&self, data: crate::TransferFileEventData) {}
+    fn developer_exception(&self, data: crate::DeveloperExceptionEventData) {}
+    fn developer_exception_with_value(&self, data: crate::DeveloperExceptionWithValueEventData) {}
```

Comment:
Minor:
```suggestion
    fn event_init(&self, data: crate::InitEventData) {}
    fn event_transfer_intent(&self, _: crate::TransferIntentEventData) {}
    fn event_transfer_start(&self, _: crate::TransferStartEventData) {}
    fn event_transfer_end(&self, _: crate::TransferEndEventData) {}
    fn event_transfer_file(&self, _: crate::TransferFileEventData) {}
    fn developer_exception(&self, _: crate::DeveloperExceptionEventData) {}
    fn developer_exception_with_value(&self, _: crate::DeveloperExceptionWithValueEventData) {}
```
and remove the `allow(unused_`